core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -83,6 +87,51 @@ class LogManagerTest {
     log.appendAsLeader(TestUtils.singletonRecords("test".getBytes()), leaderEpoch = 0)
   }
 
+  /**
+   * Tests that all internal futures are completed before LogManager.shutdown() returns to the
+   * caller during error situations.
+   */
+  @Test
+  def testHandlingExceptionsDuringShutdown(): Unit = {
+    logManager.shutdown()
```

Comment:
Hmm, do we need this given that we do this in tearDown() already?

Reply:
Yeah this explicit shutdown is needed to:
1) Re-create a new `LogManager` instance with multiple `logDirs` for this test. This is different from the default one provided in `setUp()`.
2) Help do some additional checks post shutdown (towards the end of this test).

Reply:
Thinking about it again, you are right. I have eliminated the need for the `shutdown()` now by using a `LogManager` instance specific to the test.